misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/test/StartService.java

```diff
@@ -198,80 +199,66 @@ private String getServiceURL(ServiceInfo service) {
         if (annotationInfo == null) {
             return HttpConnectionManager.getInstance().getDefaultListenerConfiugrationSet();
         }
-    
+
         //key - listenerId, value - listener config property map
         Set<ListenerConfiguration> listenerConfSet = new HashSet<>();
-    
+
         extractBasicConfig(annotationInfo, listenerConfSet);
         extractHttpsConfig(annotationInfo, listenerConfSet);
-    
+
         if (listenerConfSet.isEmpty()) {
             listenerConfSet = HttpConnectionManager.getInstance().getDefaultListenerConfiugrationSet();
         }
-    
+
         return listenerConfSet;
     }
-    
+
     private void extractBasicConfig(AnnAttachmentInfo configInfo, Set<ListenerConfiguration> listenerConfSet) {
         AnnAttributeValue hostAttrVal = configInfo.getAttributeValue(HttpConstants.ANN_CONFIG_ATTR_HOST);
         AnnAttributeValue portAttrVal = configInfo.getAttributeValue(HttpConstants.ANN_CONFIG_ATTR_PORT);
         AnnAttributeValue keepAliveAttrVal = configInfo.getAttributeValue(HttpConstants.ANN_CONFIG_ATTR_KEEP_ALIVE);
         AnnAttributeValue transferEncoding = configInfo.getAttributeValue(
                 HttpConstants.ANN_CONFIG_ATTR_TRANSFER_ENCODING);
         AnnAttributeValue chunking = configInfo.getAttributeValue(HttpConstants.ANN_CONFIG_ATTR_CHUNKING);
-        
+
         ListenerConfiguration listenerConfiguration = new ListenerConfiguration();
         if (portAttrVal != null && portAttrVal.getIntValue() > 0) {
             listenerConfiguration.setPort(Math.toIntExact(portAttrVal.getIntValue()));
-            
+
             listenerConfiguration.setScheme(HttpConstants.PROTOCOL_HTTP);
             if (hostAttrVal != null && hostAttrVal.getStringValue() != null) {
                 listenerConfiguration.setHost(hostAttrVal.getStringValue());
             } else {
                 listenerConfiguration.setHost(HttpConstants.HTTP_DEFAULT_HOST);
             }
-            
+
             if (keepAliveAttrVal != null) {
-                listenerConfiguration.setKeepAlive(keepAliveAttrVal.getBooleanValue());
+                listenerConfiguration.setKeepAliveConfig(
+                        HttpUtil.getChunkKeepAliveConfig(keepAliveAttrVal.getStringValue()));
             } else {
-                listenerConfiguration.setKeepAlive(Boolean.TRUE);
+                listenerConfiguration.setKeepAliveConfig(ChunkKeepAliveConfig.AUTO);
```

Comment:
ChunkKeepAliveConfig is very confusing and hard to understand. Shall we use separate config for keepAlive?